### DISCUSSION TITLE
Questions from a prospective user

### DISCUSSION THREAD

**Discussion by patmaddox** [helloSystem/hello#294] [General]
Hi there, I'm a web application developer who has been working on Mac since the first OS X. I am moving more from Linux to FreeBSD for server platforms, and so I'm interested in using FreeBSD as a development environment as well. Some research has led me to helloSystem. I've read as much as I can of the docs, discussions, and wiki, and have a few questions.

How would you describe the differences between helloSystem and GhostBSD? I assume you became reasonably familiar with GhostBSD before starting on helloSystem. From what I can tell, a key difference is GhostBSD is a fork of FreeBSD. helloSystem is... a set of packages that run on FreeBSD? I can't quite tell. There are a lot of repos under the helloSystem org, and the one I would expect to be the main one (this one, hello), doesn't appear to have any source code in it. So I'm not sure what is the entry point for helloSystem.

What do you mean by "we may not always be interested in contributions that increase portability"? Portability to what?

Is this intended to provide the full power of FreeBSD under the hood, or be more locked down like MacOS? I ask for two reasons: 1. There's the design goal of being a read-only OS, like macOS. So I assume that means we don't update helloSystem with `freebsd-update`? 2. One of the unwelcome technologies is configuration files - but I _want_ to edit `/etc/pf.conf`. A big part of why I want to use FreeBSD is so I have that control. This read-only, no-config-files approach appears to be in conflict with "Without lockdown. Without Big Brother. The user in full control." So perhaps I'm missing something.

Ultimately what I want is FreeBSD with a good desktop environment. So maybe that means using FreeBSD with one of the available desktop environments :)  But my research led me here, so I'm trying to get more info. I have never done any sort of OS development, so please forgive my naive questions!

**Comment by probonopd** [comment]
Hello @patmaddox, thanks for your excellent questions.

> How would you describe the differences between helloSystem and GhostBSD?

The main difference is that helloSystem is explicitly made to be welcoming to switchers, especially from the Mac. This means that we are using, e.g., a global menu, the Command key for keyboard shortcuts, and application bundles. In terms of the underlying technology, both systems are based on FreeBSD. Whereas GhostBSD has been building their own binary packages, we are trying to use the official FreeBSD packages wherever possible in helloSystem.

> From what I can tell, a key difference is GhostBSD is a fork of FreeBSD. helloSystem is... a set of packages that run on FreeBSD? 

Right now I would characterize helloSystem as "FreeBSD preconfigured as a desktop for mere mortals":

* helloSystem is an operating system that can be booted from an installable Live ISO, where we want to have an end-to-end experience that is suitable for non-technical users
* Most of the helloSystem specific components are not packaged yet in FreeBSD Ports, so the helloSystem ISO is put together from FreeBSD base and packages, plus the helloSystem specific components from GitHub
* The desktop environment, helloDesktop, is a part of helloSystem that is currently being ported to FreeBSD Ports by FreeBSD Ports maintainers

> There are a lot of repos under the helloSystem org, and the one I would expect to be the main one (this one, hello), doesn't appear to have any source code in it. So I'm not sure what is the entry point for helloSystem.

This one is mainly for project overarching topics and discussion. The actual code lives in other repositories in the helloSystem organization. The main entry point for building the helloSystem ISO is https://github.com/helloSystem/ISO/, specifically the `build.sh` script therein.

What do you mean by "we may not always be interested in contributions that increase portability"? Portability to what?

> Is this intended to provide the full power of FreeBSD under the hood, or be more locked down like MacOS?

The idea is to implement what we call our "[Three-layer UX design philosophy for Simplicity and Power](https://medium.com/@probonopd/hellosystem-three-layer-ux-design-philosophy-for-simplicity-and-power-37c95bf58398)". I hope that the linked article can answer some of your questions.

> I ask for two reasons: 1. There's the design goal of being a read-only OS, like macOS. So I assume that means we don't update helloSystem with freebsd-update?

Right now helloSystem can be run from the Live ISO, and can be "traditionally" installed to hard disk. The installed system can be updated with `freebsd-update` and `pkg upgrade`, 0.7.0 will have a GUI tool for this. We are thinking about booting the system from an immutable image file on disk, which would have advatages in terms of robustness (similar to how embedded appliances work).

> 2. One of the unwelcome technologies is configuration files - but I want to edit `/etc/pf.conf`. A big part of why I want to use FreeBSD is so I have that control. This read-only, no-config-files approach appears to be in conflict with "Without lockdown. Without Big Brother. The user in full control." So perhaps I'm missing something.

The reason that _configuration files_ are mentioned under _unwelcome technologies_ is **not** to take away control from the (power) user. All configuration files that are there can of course be customized. Even if we go the route of immutable images there will of course be a mechanism to customize the system in any way you'd like. But for the components developed for helloSystem we try to make things "just work" without the user having to experiment with 1,001 settings (which would make it impossible to test each and every combination, and make things harder to support).

> Ultimately what I want is FreeBSD with a good desktop environment. So maybe that means using FreeBSD with one of the available desktop environments :) But my research led me here, so I'm trying to get more info. I have never done any sort of OS development, so please forgive my naive questions!

Thanks for your questions. A "good desktop environment" definitely means different things to different people, but you sould get a pretty solid understanding of what I consider good user experience by reading through my [6-part article series on #LinuxUsability](https://medium.com/@probonopd/make-it-simple-linux-desktop-usability-part-1-5fa0fb369b42). In fact, helloSystem was born out of a desire to not only complain about things, but to improve them.

Let us know how you like helloSystem in case you give it a try.

**Reply by patmaddox** [reply]
Okay that is all very encouraging! The 3-layer philosophy makes a ton of sense. I'll definitely give it a try and let you know how it goes!

**Reply by grahamperrin** [reply]
> … [Three-layer UX design philosophy for Simplicity and Power](https://medium.com/@probonopd/hellosystem-three-layer-ux-design-philosophy-for-simplicity-and-power-37c95bf58398) …

![image](https://user-images.githubusercontent.com/192271/192153512-af2951a1-5200-4d29-becb-b1958fc4088b.png)

>> Which one would you want to use if all you wanted to do was switch the keyboard language?

KDE Plasma, of course. Here's the switcher: 

![image](https://user-images.githubusercontent.com/192271/192153713-a3ac2a63-8212-45fa-b833-077e442cde53.png)

What's in your article is _not_ the switcher.

**Reply by probonopd** [reply]
1. Where is that menu, how do you reach it?
2. What did you have to do so that English (UK) and English (US) show up in that menu? Where are the other languages?

Our target user experience is _one_ tool that has _all_ the available languages to choose from. Not a tool to first choose languages from and then yet another tool to switch between the selected languages.

**Reply by louies0623** [reply]
Two thing, you should know that the France, which is also in the French area, uses AZERTY, but Luxembourg and Switzerland uses QWERTZ, and Canada uses QWERTY, but there is another one that is Chinese Traditional and Simplified  , and Korean it's not on the list, there is no need to talk about the CJKTS input method, this is another matter.

**Reply by louies0623** [reply]
> Where is that menu, how do you reach it?What did you 

> have to do so that English (UK) and English (US) show up in that menu? Where are the other languages?

![image](https://user-images.githubusercontent.com/44593430/208206402-c8bc99b7-61de-4f01-a4a2-1341c6b6442f.png)
